### PR TITLE
Add an article on how to use Ansible and the Yaml Config to automate Wildfly's subsystems config.

### PR DESCRIPTION
**Pull request by rpelisse**
@bstansberry Let me know what you think about this proposal!

### CODE REVIEW

**Comment by rpelisse** [comment]
@bstansberry Any chance to get this article approved and release soon? Anything I can change to make it better?

File: _posts/2024-07-10-ansible-yaml-wildfly.adoc [hunk 1]
```diff
@@ -0,0 +1,740 @@
...
+
+In short, when designing the way WildFly's setup will be provisioned, it's important to determine what needs to be added directly to the default configuration (`standalone.xml` or `standalone-full.xml`) utilized as a base and what can be parameterized using the Yaml configuration feature, employing or not, the templating functionality of Ansible.
+
+To help make these decisions, here are a few rules of thumb to keep in mind:
+
+* Large alteration of the subsystems (adding one or several, or simply rewriting entirely the default configuration) are most likely easier to achieve by providing a modified base configuration.
+* Small changes to the subsystem configuration, adding a few, straightforward resources are most likely easy enough to implement.
+* Changes in the configuration linked to the target environments can be achieved using the templating feature of Ansible.
+* No matter what, remember the https://en.wikipedia.org/wiki/KISS_principle[KISS principe (Keep It Stupid Simple)].
```
Comment (bstansberry) [inline _posts/2024-07-10-ansible-yaml-wildfly.adoc:592]: s/principe/principle

You're nice; for us mean unprofessional people it's "Keep It Simple, Stupid!"  Your way is better for this kind of article. :)

File: _posts/2024-07-10-ansible-yaml-wildfly.adoc [hunk 2]
```diff
@@ -0,0 +1,740 @@
...
+
+Let's run again the `playbook`. As in the above example run, Ansible will notice the changed to the Yaml configuration file and consequently update the target's subsystems configuration, before restarting the server.
```
Comment (bstansberry) [inline _posts/2024-07-10-ansible-yaml-wildfly.adoc:594]: s/changed/change

File: _posts/2024-07-10-ansible-yaml-wildfly.adoc [hunk 3]
```diff
@@ -0,0 +1,740 @@
...
+
+With these first requirements in place, we now move to the deployment of our JDBC drivers and datasources.
+
+=== Deploy JDBC drivers and datasources
+
+The deployment of JDBC drivers and datasources on the target system is a somewhat more elaborated use case than the one we just saw with the messaging subsystem. Indeed, to add a JDBC driver to a WildFly server an entire module must be created, it's not just a configuration change in the `standalone.xml` that needs to be performed in an idempotent manner.
```
Comment (bstansberry) [inline _posts/2024-07-10-ansible-yaml-wildfly.adoc:600]: s/created,/created;/g

File: _posts/2024-07-10-ansible-yaml-wildfly.adoc [hunk 4]
```diff
@@ -0,0 +1,740 @@
...
+
+Fortunately, here again, the Ansible collection for WildFly does most of the heavy lifting. In fact, the default `playbook` we used already comes with the setup of two JDCB drivers:
```
Comment (bstansberry) [inline _posts/2024-07-10-ansible-yaml-wildfly.adoc:602]: s/JDCB/JDBC

File: _posts/2024-07-10-ansible-yaml-wildfly.adoc [hunk 5]
```diff
@@ -0,0 +1,740 @@
...
+      vars:
+        wildfly_driver_module_name: "{{ item.name }}"
+        wildfly_driver_version: "{{ item.version }}"
+        wildfly_driver_jar_filename: "{{ item.jar_file }}"
+        wildfly_driver_jar_url: "{{ item.url }}"
+      loop: "{{ jdbc_drivers }}"
+...
+----
+
+As shown above, the collection provides a generic role that takes care of creating the file hierarchy associated to a JDBC driver, but also download the required artifacts (jar file) along with generating the needed descriptor (module.xml). The driver's specific values are stored in the `vars.yml`, imported by Ansible when executing this `playbook`:
```
Comment (bstansberry) [inline _posts/2024-07-10-ansible-yaml-wildfly.adoc:623]: s/download/downloading

File: _posts/2024-07-10-ansible-yaml-wildfly.adoc [hunk 6]
```diff
@@ -0,0 +1,740 @@
...
+[source, yml]
+----
+postgres_driver_version: 9.4.1212
+mariadb_driver_version: 3.2.0
+jdbc_drivers:
+    - { version: "{{ postgres_driver_version }}", name: 'org.postgresql', jar_file: "postgresql-{{ postgres_driver_version }}.jar", url: "https://repo.maven.apache.org/maven2/org/postgresql/postgresql/{{ postgres_driver_version }}/postgresql-{{ postgres_driver_version }}.jar" }
+    - { version: "{{ mariadb_driver_version }}", name: 'org.mariadb', jar_file: "mariadb-java-client-{{ mariadb_driver_version }}.jar", url: "https://repo1.maven.org/maven2/org/mariadb/jdbc/mariadb-java-client/{{ mariadb_driver_version }}/mariadb-java-client-{{ mariadb_driver_version }}.jar" }
+----
+
+*Note:* _The Ansible collection for WildFly comes with a default template to generate the `module.xml` of a custom module. Obviously, this template might not be a good fit for ALL the drivers that user may have to install in a WildFly setup. For this reason, the template itself can easily be replaced by another one, providing by the user._
```
Comment (bstansberry) [inline _posts/2024-07-10-ansible-yaml-wildfly.adoc:634]: s/user/users
s/providing/provided

File: _posts/2024-07-10-ansible-yaml-wildfly.adoc [hunk 7]
```diff
@@ -0,0 +1,740 @@
...
+
+If this role ensures the modules are ready to be utilized, it does not, however, activate them. To make them available to use for datasources, we will add their definition to our Yaml configuration file:
```
Comment (bstansberry) [inline _posts/2024-07-10-ansible-yaml-wildfly.adoc:636]: s/If this/While this/g

**Review by bstansberry — CHANGES_REQUESTED** [review]
@rpelisse This is good stuff. Once published WDYT about converting this to guides for https://www.wildfly.org/guides/?

**Comment by rpelisse** [comment]
@bstansberry Sorry for the delay, I have been on sick leave for the past weeks.

I've added all your fixes (check the new commit if you want to double check). Assuming it's good for you, I can squash the 4 commits into one and we can (hopefully) release this tomorrow? (I've updated the date accordingly).

Regarding the guide, why not? I can certainly took a look at updating the existing one with some of this content.

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @rpelisse! Very nice article!

**Pull request merged**